File: src/main/java/fi/quanfoxes/parser/nodes/ContentNode.java
package fi.quanfoxes.parser.nodes;

import fi.quanfoxes.parser.Context;
import fi.quanfoxes.parser.Contextable;
import fi.quanfoxes.parser.Node;

public class ContentNode extends Node implements Contextable {

    @Override
    public Context getContext() throws Exception {
        Contextable contextable = (Contextable)first();
        return contextable.getContext();
    }
}

File: Vivid/Parser/Nodes/ContentNode.cs
public class ContentNode : Node, IResolvable
{
	public ContentNode(Position? position = null)
	{
		Position = position;
		Instance = NodeType.CONTENT;
	}

	public Node? Resolve(Context context)
	{
		foreach (var node in this) Resolver.Resolve(context, node);
		return null;
	}

	public override Type? TryGetType()
	{
		return First?.TryGetType();
	}

	public Status GetStatus()
	{
		return First != null ? Status.OK : Status.Error(Position, "Empty parenthesis are not allowed");
	}
}
